### PR TITLE
Bump `dwn-sdk-js` version `v0.2.14`

### PR DESCRIPTION
**Pull request by LiranCohen**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,8 +1,6 @@
 {
   "compilerOptions": {
-    "lib": [
-      "ESNext"
-    ],
+    "lib": ["DOM", "ES6"],
```
Comment (thehenrytsai) [inline tsconfig.json:3]: For my understanding: why the change?

Comment (LiranCohen) [reply tsconfig.json:3]: When updating `dwn-sdk-js` to the latest version I got errors like the following from an internal @web5/common module:
`Cannot find name 'BufferSource'`
`Cannot find name 'ReadableStream'`
`Cannot find name 'TransformStream'`

When I searched for those errors changing the target seemed to be the common solution. I wasn't sure about adding the "DOM" target vs "ESNext", but I saw that `dwn-server` has these two as the target and this package is used by that, so i figured that was safe.

Comment (thehenrytsai) [reply tsconfig.json:3]: I see, @frankhinek probably knows more since he worked with streams extensively.

**Review by thehenrytsai — APPROVED** [review]
